### ISSUE TITLE
Don't mix ALObjectParser and ATDD concepts?

### ISSUE THREAD

**Issue by jhoek** [dynasist/ALObjectParser#5]
Let's keep ALCodeunit clean; implement ATDD-related stuff in C# extension methods, e.g.
```csharp
// only retrieve names (cheap and sometimes all you want to know)
static IEnumerable<string> GetFeatureNames(this ALCodeunit codeunit) 

// retrieve feature details
static ALFeature GetFeature(this ALCodeunit codeunit, string featureName) 

// add new feature
static ALCodeunit AddFeature(this ALCodeunit codeunit, ALFeature feature)

// remove feature
static ALCodeunit RemoveFeature(this ALCodeunit codeunit, string featureName)
```

In ALObjectParser, allow objects and subobjects to have comments associated with them, e.g.:
- codeunits: the comment lines that preceed the properties of the codeunit, e.g. our features
- methods: the comment lines between the procedure signature and begin/var
- code lines: the comment lines immediately preceding them (?)
That way, the comments that differentiate ATDD codeunits from normal codeunits can easily be retrieved and interpreted.

Ideally, perhaps, to make sure we don't mix ALObjectParser and ATDD concepts, implement the ATDD features as separate library (with a dependency on ALObjectParser). That way, ALObjectParser cannot know about ATDD, but ATDD knows about ALObjectParser.

**Comment by martonsagi** [comment]
Yes, it's the best if ATDD parts are moved to a different library. I have implemented some parts of ATDD process to showcase the possibilities but ultimately it's a non-related feature to object parsing.

Associating comments to the codeunits, methods or method content is possible as I already have the information just don't save it anywhere. Maybe a line position or array index would also be helpful?

```
public class ALMethodBody {
    string Content { get; set; }
    IEnumerable<ALComment> Comments { get; set; }
}

public class ALComment {
    string Content { get; set; }
    int Position { get; set; }
}
```

> methods: the comment lines between the procedure signature and begin/var


Is there any specific reason why that comment goes after the method signature instead of preceding it?

**Comment by jhoek** [comment]
> Maybe a line position or array index would also be helpful?

Absolutely!

> Is there any specific reason why that comment goes after the method signature instead of preceding it?

Not sure - this is how @lvanvugt  chose to do it, I guess? Also, from a technical viewpoint, this makes the comment an actual part of the method.

**Comment by lvanvugt** [comment]
A good morning, guys. Sorry for being silent, and willbe still somewhat as I have been teaching a fool week and will do again next week and mean while supporting my main employer (like now updating our production environment). WIll step in again fully somewhere next week and answerring priority things meanwhile.

Regarding the " comment goes after the method signature" I guess that's a matter of taste. I do not like it in front. If you have better reasons to put in front .... ;-)

**Comment by martonsagi** [comment]
Well, I find it a bit strange, as comments are usually before the method definitions regardless of programming language. I don't have a specific reason why not to do it, but in this case it's best if before/after lines are also checked for comments. :)

**Comment by martonsagi** [comment]
@jhoek 

I have created a new branch to implement necessary refactoring. Anything related to Test metadata will be removed from this library to keep it clean. Should you need any reference to the original implementation, there is a [v0.0.1 tagged release](https://github.com/dynasist/ALObjectParser/releases/tag/0.0.1) for that.

**Comment by martonsagi** [comment]
I'm finished with cleaning and basic comment processing on the reading part.

Objects, Methods and MethodBody models can have `ALComments`.

Beside having a string content of Methods, body is also stored in `ALMethodBody` model. Comments below procedure definition are part of `ALMethodBody`.

```csharp
    public class ALMethodBody
    {
        public IEnumerable<string> ContentLines { get; set; }
        public string Content { get; set; }
        public IEnumerable<ALComment> Comments { get; set; }
    }
```

`ALComment` has Start/EndPos to handle multi-line comments in the future. At the moment one-line comments are processed. 

```csharp
    public class ALComment
    {
        public string Content { get; set; }
        public int StartPos { get; set; }
        public int EndPos { get; set; }
    }
```

These changes have been merged back into `master` and tagged with v0.0.2.
